docs/configuration.md

```diff
@@ -0,0 +1,316 @@
+their documentation for details.
+
+For more details on the remediation template, and the remediation CRs created
+by NHC based on the template, see [below](#remediation-resources)
+
+### EscalatingRemediations
+
+EscalatingRemediations is a list of RemediationTemplates with an order and
+timeout field. Instead of just creating one remediation CR and waiting forever
+that the node gets healthy, using this field offers the ability to to try
+multiple remediaators one after another.
+The `order` field determines the order in which the remediations are invoked
+(lower order = earlier invocation). The `timeout` field determines when the
+next remediation is invoked.
+
+There are optional features available when using escalating remediations:
+- when running into a timeout, NHC signals this to the remediator by adding
+a "remediation.medik8s.io/nhc-timed-out" annotation to the remediation CR. The
+remediator can use this to cancel its efforts.
+- The other way around, when the remediator can't remediate the node for whatever
+reason, or thinks it is done with everything, it can set a status condition of
+type "Processing" with status "False" and a current "LastTransitionTime" on the
+remediation CR. NHC will try the next remediator early then, in case the node
+doesn't get healthy within a short period of time.
+
+> **Note**
+> 
+> - This field is mutually exclusive with spec.RemediationTemplate
+> - All other notes about remediation templates made above apply here as well
+
+### UnhealthyConditions
+
+This is a list of conditions for identifying unhealthy nodes. Each condition
+has a mandatory type, status and duration.
+Type and status are compared with the node's status conditions. When they match
+for the time defined in duration, remediation will start. The list entries are
+evaluated with a logical "OR".
+
+Typically, the Ready condition is used, and the node is considered unhealthy when
+the condition status is "False" (quoted because it needs to be string value)
+or Unknown for some time. This also is the default value being set in case it's
+empty when the CR is created:
+
+```yaml
+unhealthyConditions:
+  - type: Ready
+    status: "False"
+    duration: 300s
+  - type: Ready
+    status: Unknown
+    duration: 300s
+
+```
+
+> **Warning**
+> 
+> Be careful with the value of the duration field. While it's a common desire
+> to remediate as quick as possible, a too low duration can trigger unneeded
+> remediation (which typically means reboots) in case the cluster only has a
+> short "hiccup", or when the node needs a longer time on start to get healthy.
+> For finding the best value, you need to consider the hosts reboot time, the
+> startup time of the kubernetes components and user workloads, and the
+> downtime tolerance of the user workloads.
+
+### PauseRequests
+
+When pauseRequests has at least one value set, no new remediation will be
+started, while in-flight remediations keep running.
```

Comment:
Nit: I'm not sure whether `in-flight` is a widely known term in that context, we might want to consider a minor rephrase.

Reply:
absolutely agree, I don't like that term as well, it was a copy/paste from old README leftover 👍🏼